frontend/src/pages/main/profile/components/profileInformation/ProfileInformation.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {useEffect, useState} from 'react';
+import {Card, CardContent} from "@mui/material";
+import EmailField from "../../../../../components/fields/email/EmailField";
+import NameField from "../../../../../components/fields/name/NameField";
+import PasswordField from "../../../../../components/fields/password/PasswordField";
+import {UserModification} from "../../../../../types/user";
+import {useAppDispatch, useAppSelector} from "../../../../../stores/redux/hooks";
+import {logOut, selectLoggedUser} from "../../../../../stores/redux/loggedUser/loggedUserSlice";
+import {updateUserService} from "../../../../../services/user";
+import ConfirmationButton from "../../../../../components/buttons/confirmationButton/ConfirmationButton";
+import {Buttons} from "../../../../../common/constansts/buttons";
+import {FieldsValidation} from "../../../../auth/signup/types";
+import {showNotification} from "../../../../../stores/redux/notification/notificationSlice";
+import {NotificationType} from "../../../../../types/notification";
+import {Notifications} from "../../../../../common/constansts/notifications";
+import {Titles} from "../../../../../common/constansts/titles";
+import CardTitle from "../../../../../components/typography/cardTitle/CardTitle";
+
+const initialFieldsValidation: FieldsValidation = {
+    isEmailValid: false,
+    isPasswordValid: true,
+    isNameValid: false
+};
+
+const initialUserModification: UserModification = {
+    name: "",
+    email: "",
+    password: ""
+}
+
+const ProfileInformation = () => {
+
+    const [userModification, setUserModification] = useState<UserModification>(initialUserModification);
+    const [fieldsValidation, setFieldsValidation] = useState<FieldsValidation>(initialFieldsValidation);
+    const [isValid, setIsValid] = useState<boolean>(false);
+    const loggedUser = useAppSelector(selectLoggedUser);
+    const dispatch = useAppDispatch();
+
+    const updateEmail = (email: string, isEmailValid: boolean) => {
+        setUserModification({...userModification, email});
+        setFieldsValidation({...fieldsValidation, isEmailValid});
+    }
+
+    const updateName = (name: string, isNameValid: boolean) => {
+        setUserModification({...userModification, name});
+        setFieldsValidation({...fieldsValidation, isNameValid});
+    }
+
+    const updatePassword = (password: string, isPasswordValid: boolean) => {
+        setUserModification({...userModification, password});
+        if (password === "") {
+            const validateEmptyPassword = !isPasswordValid;
```

Comment:
wat

Reply:
2 options how we can update user information: 
1) fill in all fields
2) fill in only the name and mail fields
so, for the 2 option i need to give access to button if password field if empty

Reply:
validateEmptyPassword is a name for method (verb).
Maybe validIfPasswordEmpty